src/jlapi.c

```diff
@@ -809,6 +809,28 @@ JL_DLLEXPORT uint64_t jl_cumulative_recompile_time_ns(void)
     return jl_atomic_load_relaxed(&jl_cumulative_recompile_time);
 }
 
+/**
+ * @brief Enable per-task timing.
+ */
+JL_DLLEXPORT void jl_task_metrics_enable(void)
+{
+    // Increment the flag to allow reentrant callers.
+    jl_atomic_fetch_add(&jl_task_metrics_enabled, 1);
```

Comment:
We're going to try this approach, it seems like the best path forward. 👍 
@kpamnany is going to push up some commits here. Thanks valentin!

Reply:
(jk, we didn't do this because it was too big of a change. I guess there are bootstrapping challenges relating to adding fields to the Task struct. But we'll probably make this change the next time we add more metrics)